packages/client-sdk-web/test/integration/integration-setup.ts

```diff
@@ -12,13 +14,28 @@ import {CacheClient, TopicClient, Configurations, AuthClient} from '../../src';
 import {ITopicClient} from '@gomomento/sdk-core/dist/src/clients/ITopicClient';
 import {ICacheClient} from '@gomomento/sdk-core/dist/src/clients/ICacheClient';
 
-const credsProvider = CredentialProvider.fromEnvironmentVariable({
-  environmentVariableName: 'TEST_AUTH_TOKEN',
-});
+let _credsProvider: CredentialProvider | undefined = undefined;
+
+function credsProvider(): CredentialProvider {
+  if (_credsProvider === undefined) {
+    if (isLocalhostDevelopmentMode()) {
+      _credsProvider = CredentialProvider.fromEnvironmentVariable({
+        environmentVariableName: 'TEST_AUTH_TOKEN',
+        controlEndpoint: 'https://no-controlplane-requests-allowed:9001',
+        cacheEndpoint: 'https://localhost:9001',
```

Comment:
1. I think this should be `http` and not `https`? I think you had to do some work around with tls termination otherwise?
2. port should be `9002` (for http) or `9112` (for https).
3. I got stuck big time with `localhost`. I think it has to be `0.0.0.0`.

Reply:
1. http did not work at all for me, and kenny said that it can't/won't. https works but requires some TLS setup that I documented in notion. if you are able to get it working with plaintext http that would simplify the instructions so i'd love to go that route, but i couldn't.
2. data plane port is 9001 on my box, which is pretty much a fresh clone of the repo. not sure how yours is 9002, but if it is, maybe we need to make this configurable. kenny told me 9112 is an admin port that is not relevant or useful for this task.
3. not sure how that could have happened. do you have anything weird in /etc/hosts? localhost works fine for me, and historically putting 0.0.0.0 in on the client side of stuff doesn't work for most tools in my experience :) but if 0.0.0.0 ends up working for people and localhost doesn't, we can change this

Reply:
1. http okay i thought i had it worked in the past. nbd.
2. lol yes you're right it should be 9001, i had an override in my branch.
3. Not sure. If this works on your machine then let's keep it at localhost then.